### PR TITLE
[Feature] Migrate the desc statement from the old to new parser and analyzer

### PR DESCRIPTION
**Pull request by leoyy0316**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/starrocks/issues/7285

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 122 (01.64%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|0|96|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/analysis/DescribeStmt.java|1|14|07.14%|
|:large_blue_circle:|com/starrocks/common/ErrorCode.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtAnalyzer.java [hunk 1]
```diff
@@ -111,5 +130,148 @@ String getFullDatabaseName(String db, ConnectContext session) {
...
+                    if (table.getType() == Table.TableType.OLAP) {
+                        procString += ((OlapTable) table).getBaseIndexId();
+                    } else {
+                        procString += table.getId();
+                    }
+
+                    try {
+                        node.setNode(ProcService.getInstance().open(procString));
+                    } catch (AnalysisException e) {
+                        ErrorReport.reportSemanticException(ErrorCode.ERR_BAD_PROC_NODE_ERROR, procString);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ShowStmtAnalyzer.java:198]: you can throw SemanticException

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java [hunk 2]
```diff
@@ -411,5 +412,19 @@ public Void visitCreateAnalyzeJobStatement(CreateAnalyzeJobStmt statement, Conne
             }
             return null;
         }
+
+        @Override
+        public Void visitDescTableStmt(DescribeStmt statement, ConnectContext session) {
+            String db = statement.getDb();
+            String tableName = statement.getTableName();
+            if (!GlobalStateMgr.getCurrentState().getAuth().checkTblPriv(ConnectContext.get(),
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java:420]: you can use PrivilegeChecker.checkTblPriv()

File: fe/fe-core/src/test/java/com/starrocks/analysis/DescribeStmtTest.java [hunk 3]
```diff
@@ -24,61 +24,55 @@
...
+        ctx = UtFrameUtils.createDefaultCtx();
 
-        new MockUp<ConnectContext>() {
-            @Mock
-            public ConnectContext get() {
-                return ctx;
-            }
-        };
+        String createDbStmtStr = "create database testDb;";
+        CreateDbStmt createDbStmt = (CreateDbStmt) UtFrameUtils.parseAndAnalyzeStmt(createDbStmtStr, ctx);
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/analysis/DescribeStmtTest.java:44]: parseStmtWithNewParser

File: fe/fe-core/src/test/java/com/starrocks/analysis/DescribeStmtTest.java [hunk 4]
```diff
@@ -24,61 +24,55 @@
...
-        ctx.setRemoteIP("192.168.1.1");
+import java.io.IOException;
 
-        analyzer = AccessTestUtil.fetchAdminAnalyzer(true);
-        globalStateMgr = AccessTestUtil.fetchAdminCatalog();
+public class DescribeStmtTest {
+    private static ConnectContext ctx;
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        UtFrameUtils.createMinStarRocksCluster();
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/analysis/DescribeStmtTest.java:40]: eg: 
private static StarRocksAssert starRocksAssert;
starRocksAssert = new StarRocksAssert();
starRocksAssert.withDatabase("testDb").useDatabase("testTbl");

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 68 / 118 (57.63%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|50|96|52.08%|
|:large_blue_circle:|com/starrocks/analysis/DescribeStmt.java|9|13|69.23%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|

**Comment by stephen-shelby** [comment]
run license/cla

**Review by HangyuanLiu — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 68 / 118 (57.63%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|50|96|52.08%|
|:large_blue_circle:|com/starrocks/analysis/DescribeStmt.java|9|13|69.23%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|5|5|100.00%|

**Comment by leoyy0316** [comment]
run license/cla

**Pull request closed without merging**